index.js

```diff
@@ -33,9 +39,9 @@ module.exports = function(glob, options) {
 
   // make glob absolute
   if (rootDir && glob.charAt(0) === '/') {
-    glob = path.join(rootDir, glob);
-  } else if (!isAbsolute(glob)) {
-    glob = path.resolve(cwd, glob);
+    glob = join(rootDir, glob);
+  } else if (process.platform === 'win32' || !isAbsolute(glob)) {
```

Comment:
awesome! I'm getting caught up but it looks like you got them all passing. fwiw I've been working on matching-related stuff because of this almost non-stop for the past couple of weeks. I'm going to create some issues to get feedback on a few things as they come up.